client/helpers.go

```diff
@@ -121,7 +121,12 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 			return err
 		}
 		r.RemoveKeys(td.RemoveKeys)
-		r.RemovePaths(td.RemovePaths)
+
+		removePaths := td.RemovePaths
+		if td.ClearPaths {
```

Comment:
@diogomonica 's suggesting is too dependent on order of operations. If a user wants to take an existing role that has the `""` path permissions, and restrict it to `"foo/"`, this implicit behaviour would require them to remove `""` before adding `"foo/"`.